components/SpotForm.js

```diff
@@ -0,0 +1,169 @@
+import { useState } from "react";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import { formTaskOptions } from "../helpers/formTaskOptions";
+import { formEquipmentOptions } from "../helpers/formEquipmentOptions";
+export default function SpotForm({ addSpot }) {
+  const [formData, setFormData] = useState({
+    id: Date.now(),
+    Spotname: "",
+    Address: "",
+    Tasks: [],
+    Equipment: [],
+  });
+
+  const router = useRouter();
+  const handleOptionToggle = (option, field) => {
+    setFormData((prevFormData) => {
+      const selectedOptions = prevFormData[field];
+
+      if (selectedOptions.includes(option)) {
+        return {
+          ...prevFormData,
+          [field]: selectedOptions.filter(
+            (selectedOption) => selectedOption !== option
+          ),
+        };
+      } else {
+        return {
+          ...prevFormData,
+          [field]: [...selectedOptions, option],
+        };
+      }
+    });
+  };
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [name]: value,
+    }));
+  };
+
+  const onSubmitHandler = (event) => {
+    event.preventDefault();
+    const newSpot = {
+      id: formData.id,
+      Spotname: formData.Spotname, // Updated property name
+      Address: formData.Address, // Updated property name
+      Tasks: formData.Tasks, // Updated property name
+      Equipment: formData.Equipment,
+    };
+    const existingSpots = JSON.parse(localStorage.getItem("spots")) || [];
+    const updatedSpots = [...existingSpots, newSpot];
+    localStorage.setItem("spots", JSON.stringify(updatedSpots));
+    addSpot(newSpot);
```

Comment:
```suggestion
    const updatedSpots = [...existingSpots, newSpot];
    addSpot(newSpot);
```
- you already update the spot data in your local storage with addSpot :)